### PR TITLE
Trim offset calls do not get applied to trail related layers.

### PR DESCRIPTION
**Pull request by RingerJK**
### Description
CP https://github.com/mapbox/mapbox-navigation-android/pull/5982

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,26 @@ Mapbox welcomes participation and contributions from everyone.
 #### Features
 
 #### Bug fixes and improvements
+- Fixed reroute request interruption when setting the `NavigationRerouteController` [#5950](https://github.com/mapbox/mapbox-navigation-android/pull/5950).
```
Comment (abhishek1508) [inline CHANGELOG.md:9]: Let's make sure this is not causing regression before we cherry pick this commit. 
@RingerJK

Comment (Guardiola31337) [reply CHANGELOG.md:9]: BTW release notes should be cherry-picked separately (if we finally decide to cherry-pick https://github.com/mapbox/mapbox-navigation-android/pull/5950).

Comment (Zayankovsky) [reply CHANGELOG.md:9]: I believe most of these lines were cherry-picked by mistake.

Comment (RingerJK) [reply CHANGELOG.md]: @Guardiola31337 do you mean that changelog should be fully cherry-picked from main?

Comment (Guardiola31337) [reply CHANGELOG.md]: `CHANGELOG` is always going to bring conflicts when cherry-picking. What I meant was that those lines were not relevant to this cherry-pick (those are related to the other PR #5950 and should be cherry-picked separately - along with the rest of its changes).

Comment (Zayankovsky) [reply CHANGELOG.md]: So, should there be a changelog entry for this change or not? I think there should be, but there isn't any at the moment.

Comment (Guardiola31337) [reply CHANGELOG.md]: Yeah, for some reason it was removed. The only thing that I was saying was that that other entry was not for this PR but we should keep the one from this PR 👍

**Review by abhishek1508 — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (Zayankovsky) [reply CHANGELOG.md:10]: @RingerJK you have added a wrong changelog entry.

**Review by Zayankovsky — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (RingerJK) [reply CHANGELOG.md]: @Zayankovsky rebase issue, fixed 👍

**Review by Zayankovsky — APPROVED** [review]

**Pull request merged**